android/build_boinc_arm.sh

```diff
@@ -27,20 +27,20 @@ export CFLAGS="--sysroot=$TCSYSROOT -DANDROID -DDECLARE_TIMEZONE -Wall -I$TCINCL
 export CXXFLAGS="--sysroot=$TCSYSROOT -DANDROID -Wall -I$TCINCLUDES/include -funroll-loops -fexceptions -O3 -fomit-frame-pointer -fPIE -march=armv7-a"
```

Comment:
What's the reasoning behind not improving the contents of `CXXFLAGS`, `LDFLAGS` and `GDB_CFLAGS` in the same way (in all scripts)? I'm not sure the outer quotation marks are effective for the contained inner variables like in `-I$TCINCLUDES/include` as well. Are they?

Reply:
Interesting question actually… I so routinely ignore this issue I didn't have an imediate answer for this.

Turns out the answer is twofold:

  * As long as you use `autotools` there is absolutely nothing you can do here – any attempt at adding quotation or similar will make things worse! And [`cmake` is in no way better](https://gitlab.kitware.com/cmake/cmake/blob/master/Modules/CMakeCInformation.cmake#L103) if I understand this code correctly.
  * I also checked `meson` and [it uses Python's `shlex.split`](https://github.com/mesonbuild/meson/blob/ae0e469b0692862dae10ceb02799feb3f117ce26/mesonbuild/environment.py#L924), which means it would support quoting the values correctly.

TL;DR: Yes this sucks and there is no way to fix it currently.